pkgs/development/tools/okteto/default.nix

```
@@ -0,0 +1,45 @@
+{ lib, buildGoModule, fetchFromGitHub, installShellFiles }:
+
+buildGoModule rec {
+  pname = "okteto";
+  version = "2.3.0";
+
+  src = fetchFromGitHub {
+    owner = "okteto";
+    repo = "okteto";
+    rev = version;
+    sha256 = "sha256-4Ky2pRpOYtfeEvGu5chfHKogAN5nmEN3sgaeHJhe+3w=";
+  };
+
+  postPatch = ''
+    # Disable some tests that need file system & network access.
+    find cmd -name "*_test.go" | xargs rm -f
+    rm -f pkg/analytics/track_test.go
+  '';
+
+  vendorSha256 = "sha256-XT/ZLydN1oeuRupD3gjvY6+hOB/Lq5CQwhfr9/iT7JI=";
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/okteto/okteto/pkg/config.VersionString=${version}"
+  ];
+
+  tags = [ "osusergo" "netgo" "static_build" ];
+
```

```suggestion

  preCheck = ''
    export HOME=$(mktemp -d)
  '';

```
fixes ``time="2022-05-26T12:02:29Z" level=fatal msg="failed to create /homeless-shelter/.okteto: mkdir /homeless-shelter: permission denied"``